src/foamadapter/framework/conditions.py

```diff
@@ -0,0 +1,41 @@
+"""
+Unified Condition class with logical operators.
+"""
+
+from typing import Callable
```

Comment:
Use TypeVar, see  https://docs.python.org/3/library/typing.html#generics

```suggestion
from typing import Callable,  TypeVar

Condition =  TypeVar('Condition')  
```

Reply:
the recommend approach is to handle it after 3.7+ with:

from __future__ import annotations

after 3.10:

from typing import Self

def __and__(self, other: Self) -> Self:

is a possiblity